TCMSApp/src/server/app.js

```diff
@@ -3,25 +3,56 @@
 
 var express = require('express');
 var app = express();
+var methodOverride = require('method-override');
+var mongoose = require('mongoose');
+var restify = require('express-restify-mongoose');
```

Comment:
you've definitely forgot to add some modules to package.json.
- express-restify-mongoose
- express-mongoose-docs